### PR TITLE
REF: de-duplicate pad/backfill code

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/_libs/algos.pyx [hunk 1]
```diff
@@ -794,68 +794,14 @@ def backfill(ndarray[algos_t] old, ndarray[algos_t] new, limit=None) -> ndarray:
...
-            if fill_count >= lim:
-                continue
-            fill_count += 1
-            values[i] = val
-            mask[i] = prev_mask
-        else:
-            fill_count = 0
-            val = values[i]
-            prev_mask = mask[i]
+    return pad_inplace(values[::-1], mask[::-1], limit=limit)
```
Comment (simonjayhawkins) [inline pandas/_libs/algos.pyx:798]: do we need the return?

Comment (jbrockmendel) [reply pandas/_libs/algos.pyx:798]: nope, will remove

**Comment by jreback** [comment]
lgtm if you can address comment.

**Comment by jbrockmendel** [comment]
updated + greenish

**Pull request merged**